pennylane/workflow/construct_batch.py

```diff
@@ -336,16 +438,22 @@ def circuit(x):
          expval(X(0) + Y(0))]
 
     """
+    _validate_level(level)
+    is_torch_layer = type(qnode).__name__ == "TorchLayer"
+    has_shots_param = "shots" in inspect.signature(qnode.func).parameters
+    default_shots = qnode._shots  # pylint:disable=protected-access
+
+    user_program = qnode.transform_program
+    num_user_transforms = len(user_program)
 
     def batch_constructor(*args, **kwargs) -> tuple[QuantumScriptBatch, PostprocessingFn]:
         """Create a batch of tapes and a post processing function."""
-        if "shots" in inspect.signature(qnode.func).parameters:
-            shots = qnode.device.shots
+        if has_shots_param:
+            shots = default_shots
         else:
-            shots = kwargs.pop("shots", qnode.device.shots)
+            shots = kwargs.pop("shots", default_shots)
```

Comment:
Let's make sure we're following the same logic as is in the qnode now 

https://github.com/PennyLaneAI/pennylane/blob/7db6e767cabd8f91ef89c9dbd3a031a140fba9af/pennylane/workflow/qnode.py#L854

We should have the same priority as the qnode. So if shots is provided, but we already opted into the new system, we should still be using `default_shots`.